### PR TITLE
[dev] Monorepo: relax webpack TypeScript errors (turn into warnings)

### PR DESCRIPTION
**Pull request by kalessil**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Clean extraction of PoC work here: https://github.com/woocommerce/woocommerce/pull/47486.
In this PR, we relax TS errors in the admin package.

### How to test the changes in this Pull Request:

- CI checks shall pass
- Sample run with an error turned into warning: [here](https://github.com/woocommerce/woocommerce/actions/runs/11323064045/job/31485031201?pr=51993#step:3:802)

### CODE REVIEW

File: packages/js/internal-style-build/index.js [hunk 1]
```diff
@@ -8,6 +8,26 @@ const RemoveEmptyScriptsPlugin = require( 'webpack-remove-empty-scripts' );
...
+/**
+ * https://github.com/woocommerce/woocommerce/pull/47486: do not break build if TS emits errors - React 18 migration related.
+ * The solution is based https://github.com/TypeStrong/fork-ts-checker-webpack-plugin/issues/232#issuecomment-1322651312.
+ *
+ * @deprecated intended for React version migration and can be dropped from exports any time without further notice.
+ */
+class TypeScriptWarnOnlyWebpackPlugin {
+	apply( compiler ) {
+		const hooks = ForkTsCheckerWebpackPlugin.getCompilerHooks( compiler );
+		hooks.issues.tap( 'TypeScriptWarnOnlyWebpackPlugin', ( issues ) =>
```
Comment (kalessil) [inline packages/js/internal-style-build/index.js:30]: Note: we expect to filter specific TS error codes here.

File: plugins/woocommerce/client/admin/client/core-profiler/index.tsx [hunk 2]
```diff
@@ -1582,8 +1582,7 @@ export const CoreProfilerController = ( {
 	const augmentedStateMachine = useMemo( () => {
 		// When adding extensibility, this is the place to manipulate the state machine definition.
 		return coreProfilerStateMachineDefinition.provide( {
-			// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-			// @ts-ignore -- there seems to be a flaky error here - it fails sometimes and then not on recompile, will need to investigate further.
+			// TODO: revert the change - intention is to demo for relaxing TS errors in webpack.
```
Comment (kalessil) [inline plugins/woocommerce/client/admin/client/core-profiler/index.tsx]: It provokes a TypeScript error, but the build should pass.

Update: see [here](https://github.com/woocommerce/woocommerce/actions/runs/11291966408/job/31407042445?pr=51993#step:3:611).

File: packages/js/internal-style-build/index.js [hunk 3]
```diff
@@ -8,6 +8,26 @@ const RemoveEmptyScriptsPlugin = require( 'webpack-remove-empty-scripts' );
...
 const StyleAssetPlugin = require( './style-asset-plugin' );
 
+/**
+ * @deprecated intended for React version migration and can be dropped from exports any time without further notice.
+ */
+const ForkTsCheckerWebpackPlugin = require( 'fork-ts-checker-webpack-plugin' );
+
+/**
+ * https://github.com/woocommerce/woocommerce/pull/47486: do not break build if TS emits errors - React 18 migration related.
+ * The solution is based https://github.com/TypeStrong/fork-ts-checker-webpack-plugin/issues/232#issuecomment-1322651312.
```
Comment (samueljseay) [inline packages/js/internal-style-build/index.js]: nitpick  "is based on"

Comment (kalessil) [reply packages/js/internal-style-build/index.js]: Thanks, updated in a8bcdffeebc1f02c4145bfd73317c0d11912a3c8.

**Pull request merged**